catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardImpl.java

```diff
@@ -157,6 +157,9 @@ public MetacardImpl(Metacard metacard, MetacardType type) {
             throw new IllegalArgumentException(
                     MetacardType.class.getName() + " instance should not be null.");
         }
+        if (metacard.getSourceId() != null) {
```

Comment:
I doubt we need it at all. This is a clone constructor. Not setting it is the same as setting it to null.